src/Model/test.js

```diff
@@ -208,6 +209,66 @@ describe('CRUD', () => {
     expect(nextObserver).toHaveBeenCalledTimes(1)
     expect(completionObserver).toHaveBeenCalledTimes(1)
   })
+  it('can destroy a record and its children permanently', async () => {
+    const { database, projects, tasks, comments } = mockDatabase()
+
+    const project = await projects.create(mock => {
+      mock.name = 'foo'
+    })
+
+    const task = await tasks.create(mock => {
+      mock.projectId = project.id
+    })
+
+    const comment = await comments.create(mock => {
+      mock.taskId = task.id
+    })
+
+    database.adapter.batch = jest.fn()
+    const spyBatchDB = jest.spyOn(database, 'batch')
+
+    const spyOnPrepareDestroyPermanentlyProject = jest.spyOn(project, 'prepareDestroyPermanently')
+    const spyOnPrepareDestroyPermanentlyTask = jest.spyOn(task, 'prepareDestroyPermanently')
+    const spyOnPrepareDestroyPermanentlyComment = jest.spyOn(comment, 'prepareDestroyPermanently')
+
+    await project.experimentalDestroyPermanently()
+
+    expect(spyOnPrepareDestroyPermanentlyProject).toHaveBeenCalledTimes(1)
```

Comment:
btw I think we're missing tests of prepareDestroyPermanently and prepareMarkAsDeleted -- I must have missed that during previous CRs (or maybe it's located somewhere else) -- can you check it?

Reply:
there is a test for `prepareDestroyPermanently` https://github.com/Nozbe/WatermelonDB/blob/dcefc3184fcb93dc51b189586a49ad3c11346fdc/src/Model/test.js#L189 , but I haven't seen a test for `prepareMarkAsDeleted` either. I'll add one.

Reply:
added